docs/SERVER-API.md

```diff
@@ -0,0 +1,214 @@
+This document details the API offered by the server to clients. Each method description contains:
+- The method's name
+- The method's purpose and intended effect on the server and/or the appearance of the screen rendered by zellij.
+- The method's parameters
+- The response that the client can expect to receive, including any possible error messages. N.B. Response does not detail side-effects of the request, in particular changes in what's rendered on screen.
+
+The document does not describe implementation details on the server side and is intended for use by those implementing new zellij clients.
+
+
+## Basic controls
+
+### Write input
+
+**Purpose:** Writes a set of bytes to the terminal, not to be interpeted by zellij.
+
+**Parameters:** Vector of bytes to be written.
+
+**Response:** None.
+
+### Switch input mode
+@@@Is this part of the API or a client detail?
+
+**Purpose:** Switches between different zellij [input modes](./TERMINOLOGY.md#input_modes).
+
+**Parameters:** Which input mode to switch to.
+
+**Response:** None.
+
+### Scroll up
+
+**Purpose:** Scrolls up (back) in the screen buffer of the focus [pane](./TERMINOLOGY.md#pane).
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Scroll down
+
+**Purpose:** Scrolls down (forwards) in the screen buffer of the focus [pane](./TERMINOLOGY.md#pane).
+
+**Parameters:** None.
+
+**Response:** None.
+
+### Terminal resize
+
+**Purpose:** Indicates that the terminal has been resized and so the layout should be redrawn.
+
+**Parameters:** New size of the layout.
+
+**Response:** @@@do we have a lock here?
+
+### Open file
+
+**Purpose:** Indicates that a new terminal should be spawned.
+
+**Parameters:** File name (optional) - the name of a file to be opened, using a terminal application determined by first the `EDITOR` and (if `EDITOR` is not set) `VISUAL` environment variables.
+
+**Response:** @@@do we have a lock here?
```

Comment:
This feels like an odd API. Do we actually just want the "new client" API below to optionally take some config to open a layout, with each pane in the layout optionally taking a filename to be opened using the described heuristic?

Reply:
So are you suggesting that this should only be possible through a layout?